### ISSUE TITLE
backend-core typescript error

### ISSUE THREAD

**Issue by jtparrett** [clerkinc/javascript#205]
A typescript error is being thrown from @clerk/backend-core when running `tsc --noEmit` on my project which uses @clerk/remix

```
node_modules/@clerk/backend-core/src/api/utils/RestClient.ts:77:7 - error TS2322: Type 'object | undefined' is not assignable to type 'Record<string, unknown> | undefined'.
  Type 'object' is not assignable to type 'Record<string, unknown>'.
    Index signature for type 'string' is missing in type '{}'.

77       body,
         ~~~~

  node_modules/@clerk/backend-core/src/api/utils/RestClient.ts:31:5
    31     body?: Record<string, unknown>;
           ~~~~
    The expected type comes from property 'body' which is declared here on type '{ method: "GET" | "POST" | "PATCH" | "DELETE" | "PUT"; authorization: string; contentType: string; userAgent: string; body?: Record<string, unknown> | undefined; }'
```

https://github.com/clerkinc/javascript/blob/3c73b5ac4a3de34011a1f05852681b19ab50aa2c/packages/backend-core/src/api/utils/RestClient.ts#L77

**Comment by jtparrett** [comment]
@igneel64 any update on this? Or any suggestion to get around it? Currently have a failing pipeline 🥲

**Comment by igneel64** [comment]
Hey @jtparrett , for the time being until we ship a fix, could you try setting the [skipLibCheck](https://www.typescriptlang.org/tsconfig#skipLibCheck) option to true ?

Something like :
```ts
{
  "compilerOptions": {
    "skipLibCheck": true,
  },
}
```

**Comment by jtparrett** [comment]
@igneel64 Thanks for this. I actually already have skipLibCheck enabled. Here's my full tsconfig: 

```
{
  "include": ["remix.env.d.ts", "**/*.ts", "**/*.tsx"],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2019"],
    "isolatedModules": true,
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "target": "ES2019",
    "strict": true,
    "baseUrl": ".",
    "paths": {
      "~/*": ["./app/*"]
    },
    "skipLibCheck": true,

    // Remix takes care of building everything in `remix build`.
    "noEmit": true
  }
}
```

Do you have an ETA For when a fix might be available?

**Comment by igneel64** [comment]
@jtparrett 

Can you please give me some information about the project e.g. the TypeScript version?
Can you reproduce it at https://github.com/clerkinc/remix-auth-starter with `"skipLibCheck": true` ?

**Comment by jtparrett** [comment]
@igneel64 i've opened a PR with a hacky hot fix, hopefully it highlights the issue a little clearer.
https://github.com/clerkinc/javascript/pull/208

**Comment by jtparrett** [comment]
Issue fixed in @clerk/remix v0.3.13 🙏 🎉

**Comment by igneel64** [comment]
@jtparrett Yeap thanks again for the contribution :)
